cpp/src/sampling/random_walks_impl.cuh

```diff
@@ -70,6 +72,18 @@ struct sample_edges_op_t {
   }
 };
 
+template <typename vertex_t, typename bias_t>
+struct e_bias_op_t {
+
+  raft::device_span<bias_t const> vertex_weight_sum{};
+
+  __device__ bias_t
+  operator()(vertex_t src, vertex_t, thrust::nullopt_t, thrust::nullopt_t, bias_t weight) const
+  {
```

Comment:
`vertex_weight_sum[src]` won't work in multi-GPU. Each GPU holds roughly V/P (V: # vertices, P: # GPUs) weight values, not the entire V.

You need to use `edge_src_property_t`. 


See https://github.com/rapidsai/cugraph/blob/branch-24.08/cpp/src/link_analysis/pagerank_impl.cuh#L233

https://github.com/rapidsai/cugraph/blob/branch-24.08/cpp/src/link_analysis/pagerank_impl.cuh#L264

and


https://github.com/rapidsai/cugraph/blob/branch-24.08/cpp/src/link_analysis/pagerank_impl.cuh#L279